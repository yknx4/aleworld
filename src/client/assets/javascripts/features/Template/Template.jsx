import React,{Component, PropTypes} from 'react';
import 'styles/components/Template.scss';

import {
  MainContainer,
} from 'components/layout';

import Header from './Header';
import Footer from './Footer';

export default class Template extends Component {
    render() {
      const {children} = this.props;
        return (
          <div>
            <Header />
            <MainContainer>
              {children}
              <Footer />
            </MainContainer>
          </div>
        );
    }
}

Template.propTypes = {
  children: PropTypes.node.isRequired,
};
