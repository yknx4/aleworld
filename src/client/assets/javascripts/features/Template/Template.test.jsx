import React from 'react';
import { shallow } from 'enzyme';
import Template from './Template';

describe('Template', () => {
  let shallowComponent;
  const children = <p> papi </p>;

  beforeEach(() => {
    const component = <Template >{children}</Template>;
    shallowComponent = shallow(component);
  });

  it('should render without throwing an error', function() {
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should be selectable by tag', function() {
    expect(shallowComponent.is('div')).toBe(true);
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render its children', () => {
    expect(shallowComponent.contains(children)).toBe(true);
    expect(shallowComponent).toMatchSnapshot();
  });

});
