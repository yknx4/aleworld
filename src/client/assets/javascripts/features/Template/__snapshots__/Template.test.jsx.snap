// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template should be selectable by tag 1`] = `
<div>
  <Header />
  <MainContainer>
    <p>
       papi 
    </p>
    <Footer />
  </MainContainer>
</div>
`;

exports[`Template should render its children 1`] = `
<div>
  <Header />
  <MainContainer>
    <p>
       papi 
    </p>
    <Footer />
  </MainContainer>
</div>
`;

exports[`Template should render without throwing an error 1`] = `
<div>
  <Header />
  <MainContainer>
    <p>
       papi 
    </p>
    <Footer />
  </MainContainer>
</div>
`;
